dashboard/app/models/pd/application/teacher_application.rb

```diff
@@ -84,11 +84,13 @@ class TeacherApplication < ApplicationBase
 
     has_many :emails, class_name: 'Pd::Application::Email', foreign_key: 'pd_application_id'
 
+    before_validation :set_course_from_program, if: -> {form_data_changed?}
+    before_validation :set_total_course_hours, if: -> {form_data_changed?}
+    before_validation :update_user_school_info!, if: -> {form_data_changed?}
```

Comment:
I don't have a strong opinion of if these should go before_validation or before_save ... the only one that needs to be before_validation is `:set_course_from_program` because of the `validates course` line. Any opinions are welcome